### PR TITLE
fix(Modal): Allow components as modal heading

### PR DESCRIPTION
**Pull request by tombrowndev**
Add support to supply a component as the Model Label.

Particularly useful for passing in SVG Components.

### CODE REVIEW

File: src/components/Modal/Modal.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ export default class Modal extends Component {
     onRequestClose: PropTypes.func,
     id: PropTypes.string,
     modalHeading: PropTypes.string,
-    modalLabel: PropTypes.string,
+    modalLabel: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
```
Comment (asudoh) [inline src/components/Modal/Modal.js]: Did you mean `PropTypes.node` instead...?

Comment (tombrowndev) [reply src/components/Modal/Modal.js]: Thats an even better solution, thanks!

**Comment by marijohannessen** [comment]
Thanks for fixing this @tombrowndev ! There's a tiny merge conflict that needs to be resolved before we can merge it in 👍

**Comment by joshblack** [comment]
Tried to merge in with what I thought would be the resolution, feel free to revert if that's not the case!

**Pull request merged**